overloads/vector.h

```diff
@@ -62,6 +62,29 @@ vector<T> operator + (T a, const vector<T> &b) {
     return arr;
 }
 
+// Subtraction
+template<typename T>
+vector<T> operator - (vector<T> &a, const vector<T> &b) {
+    assert("Tensors are not of the same size !" && a.size() == b.size());
+    vector<T> arr;
+    for(int i = 0;i<a.size();i++) {
+        T prod = a[i]-b[i];
+        arr.push_back(prod);
+    }
+    return arr;
+}
+
+// Scalar Subtraction
+template<typename T>
+vector<T> operator - (T a, const vector<T> &b) {
```

Comment:
Should the arguments order be reversed? We would usually subtract a scalar from a vector, not vice-versa

Reply:
Sometimes we need to do 1 - x. Or 1+ x, the sigmoid operation requires this.

Reply:
For example the backProp of sigmoid needs f*(1-f).